src/EditorFeatures/Core/Implementation/EditAndContinue/EditAndContinueSaveFileCommandHandler.cs

```diff
@@ -36,14 +38,13 @@ void IChainedCommandHandler<SaveCommandArgs>.ExecuteCommand(SaveCommandArgs args
 
             if (Workspace.TryGetWorkspace(textContainer, out var workspace))
             {
-                var encService = workspace.Services.GetService<IEditAndContinueWorkspaceService>();
-                if (encService != null)
+                var documentId = workspace.GetDocumentIdInCurrentContext(textContainer);
+                if (documentId != null)
                 {
-                    var documentId = workspace.GetDocumentIdInCurrentContext(textContainer);
-                    if (documentId != null)
-                    {
-                        encService.OnSourceFileUpdated(documentId);
-                    }
+                    var proxy = new RemoteEditAndContinueServiceProxy(workspace);
+
+                    // fire and forget
+                    _ = Task.Run(() => proxy.OnSourceFileUpdatedAsync(documentId, CancellationToken.None));
```

Comment:
so... we likely (outside of this PR) need to have a conversation on stuff like this.  tehre are def concerns i have.  

1. what happens with exceptions here?  does this trigger the unhandled exception codepaths?
2. this pattern will not work for integrationtests/asyncwaiters.  this should likely take in the appropriate integration test IAsyncListener stuff and have that be threaded through here.  that way you could actually write an integration test that saves, then waits for EnC work to finish, then validates the result.

@jasonmalinowski for thoughts on how to best code up teh concept of fire-and-forget in the IDE.

Reply:
Good points. Filed https://github.com/dotnet/roslyn/issues/47469
It'd be good to have an established pattern and helpers for fire and forget that are used consistently.